### PR TITLE
Upgrade Golang Version to 1.21.6

### LINKED ISSUE #3601: Bump Golang Version to go1.20.12
There are security vulnerabilities in our current Golang version 1.20.10 that are fixed in 1.20.12 https://security-tracker.debian.org/tracker/CVE-2023-45285.

So any instance of 1.20.10 bump it to 1.20.12. This also means that for any examples, we'll also need to bump that version number again, and redeploy as well.

Useful resources:

Github Code Search: https://github.com/search?q=repo%3Agoogleforgames%2Fagones%201.20.10&type=code
A past update: https://github.com/googleforgames/agones/pull/3137

### PR DESCRIPTION
**Pull request by Kalaiselvi84**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug

/kind cleanup
> /kind documentation
> /kind feature
> /kind hotfix
> /kind release

**What this PR does / Why we need it**:

**Which issue(s) this PR fixes**:

Closes #3601 

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by Kalaiselvi84** [comment]
The `make cloud-build` command has succeeded for all the necessary example images(autoscaler-webhook, crd-client, supertuxkart, xonotic, and simple-game-server) in my project. The files `go.mod` and `go.sum` have been updated only for the autoscaler-webhook image. 

Please confirm if this looks good to you to push these new images to production. I will commit the remaining file updates in this PR after the push.

**Comment by markmandel** [comment]
LGTM! @igooch you got anything?

**Comment by igooch** [comment]
We should update the mod files as well for the main repo and the examples `~/agones$ go mod edit -go=1.21`. You might need to update your machine and IDE  Go versions to 1.21 in order to run `go mod tidy` after the change to 1.21 in the go.mod files.

**Comment by markmandel** [comment]
> We should update the mod files as well for the main repo and the examples `~/agones$ go mod edit -go=1.21`. You might need to update your machine and IDE Go versions to 1.21 in order to run `go mod tidy` after the change to 1.21 in the go.mod files.

Good call 👍🏻

**Comment by Kalaiselvi84** [comment]
Installed the Go v1.21.6. and updated the mod files using `go mod edit -go=1.21` and `go mod tidy`.

**Comment by igooch** [comment]
We should update for the go.mod files in the examples/ directories as well.

**Comment by Kalaiselvi84** [comment]
> We should update for the go.mod files in the examples/ directories as well.

I used the same commands but mod files haven't modified. Am I missing anything?

~/Desktop/agones/examples$ go mod edit -go=1.21
~/Desktop/agones/examples$ go mod tidy

**Comment by Kalaiselvi84** [comment]
Following commands help to update the subdirectories of mod and sum files:

agones/examples$ **find . -name 'go.mod' -execdir go mod edit -go=1.21 \;**
agones/examples$ **find . -name 'go.mod' -execdir go mod tidy \;**

**Comment by igooch** [comment]
Test failure will likely be fixed with #3603.

File: examples/autoscaler-webhook/Makefile [hunk 1]
```diff
@@ -31,9 +31,9 @@ project_path := $(dir $(mkfile_path))
 root_path := $(realpath $(project_path)/../..)
 
 ifeq ($(REPOSITORY),)
-	autoscaler_webhook_tag := autoscaler-webhook:0.10
+	autoscaler_webhook_tag := autoscaler-webhook:0.11
```
Comment (markmandel) [inline examples/autoscaler-webhook/Makefile:34]: Note: As a separate PR, will need to update the e2e tests with the new version once published.

https://github.com/googleforgames/agones/blob/eb4035e1f78e7a8e0c147a013a5405f904ab9899/test/e2e/fleetautoscaler_test.go#L647

Same with simple-game-server.

Although there seem to be so many changes in PRs recently, might be better to make that change to the e2e tests once they all finish landing.

**Review by markmandel — COMMENTED** [review]
This LGTM. @igooch anything else on your end?

File: examples/autoscaler-webhook/Makefile (hunk 1 above)
Comment (Kalaiselvi84) [reply examples/autoscaler-webhook/Makefile:34]: If this looks good, I will push the images to prod and then update other files(e2e tests) in this same PR, tomorrow. WDYT?

Comment (Kalaiselvi84) [reply examples/autoscaler-webhook/Makefile:34]: Friendly reminder, does this look good to you, @igooch?

Comment (Kalaiselvi84) [reply examples/autoscaler-webhook/Makefile:34]: @igooch, Please feel free to merge this PR if the changes look good to you.😀

**Review by igooch — APPROVED** [review]
LGTM

**Pull request merged**